CONTRIBUTING.md

```diff
@@ -0,0 +1,136 @@
+2. Run `gem install rails` to get the most recent rails release
+3. Install the node version defined in the `.node-version` file with the node
+   version manager of your choice
+4. Install yarn v1
+
+To run tests you'll also need to install
+
+- [PostgreSQL](https://www.postgresql.org/)
+- [Chromedriver](https://chromedriver.chromium.org/)
+
+## Architecture
+
+This project works by hooking into the standard Rails
+[application templates](https://guides.rubyonrails.org/rails_application_templates.html)
+system, with some caveats. The entry point is the
+[template.rb](https://github.com/ackama/rails-template/blob/main/template.rb)
+file in the root of this repository.
+
+Normally, Rails only allows a single file to be specified as an application
+template (i.e. using the `-m <URL>` option). To work around this limitation, the
+first step this template performs is a `git clone` of the
+`ackama/rails-template` repository to a local temporary directory.
+
+This temporary directory is then added to the `source_paths` of the Rails
+generator system, allowing all of its ERb templates and files to be referenced
+when the application template script is evaluated.
+
+Rails generators are very lightly documented; what you'll find is that most of
+the heavy lifting is done by [Thor](http://whatisthor.com/). Thor is a tool that
+allows you to easily perform command line utilities. The most common methods
+used by this template are Thor's `copy_file`, `template`, and `gsub_file`. You
+can dig into the well-organized and well-documented
+[Thor source code](https://github.com/erikhuda/thor) to learn more. If any file
+finishes with `.tt`, Thor considers it to be a template and places it in the
+destination without the extension `.tt`.
+
+## Tests
+
+this template is tested by building a set of apps with different configuration
+and checking each passes the templated linters and tests.
+
+the config is in ci/configs/\*.yml, and is run by ci/bin/build-and-test
+
+```bash
+# create new rails app in tmp/builds/enterprise using ci/configs/react.yml as
+$ CONFIG_PATH="ci/configs/react.yml" APP_NAME="enterprise" ./ci/bin/build-and-test
+```
+
+To add another tested configuration
+
+1. add a file in ci/configs
+2. add it to the lists in .github/workflows/ci.yml
+
+## Code style
+
+Rubocop is configured for this repo and is run as part of CI. Run rubocop
+locally via the usual method:
+
+```
+$ bundle install
+$ bundle exec rubocop # optionally adding -a or -A for autofixes
+```
+
+Prettier is used to manage the style for json, js/jsx/ts/tsx, css/scss, and
+md/markdown files, and is run as part of CI. Run prettier localy via yarn
+
+```
+$ yarn install
+$ yarn run format-check # or `yarn run format-fix` for autofixes
+```
+
+## Managing your branch
+
+- Use well-crafted commit messages, providing context if possible.
+- Squash "WIP" commits and remove merge commits by rebasing your branch against
+  `main`. We try to keep our commit history as clean as possible.
```

Comment:
Since we merge by squashing, I think we should be saying roughly the opposite: while a tidy commit history is useful (especially for large changes), it doesn't have to be perfect since we'll squash everything into one, but what we do want is a meaningful PR title & description.